client/my-sites/sharing/buttons/preview-action.jsx

```diff
@@ -1,52 +1,55 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classNames = require( 'classnames' );
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+import { omit, noop } from 'lodash';
 
-module.exports = React.createClass( {
-	displayName: 'SharingButtonsPreviewAction',
-
-	propTypes: {
-		active: React.PropTypes.bool,
-		position: React.PropTypes.oneOf( [
+export default class SharingButtonsPreviewAction extends Component {
+	static propTypes = {
+		active: PropTypes.bool,
+		position: PropTypes.oneOf( [
 			'top-left',
 			'top-right',
 			'bottom-left',
 			'bottom-right'
 		] ),
-		icon: React.PropTypes.string,
-		onClick: React.PropTypes.func
-	},
+		icon: PropTypes.string,
+		onClick: PropTypes.func
+	};
+
+	static defaultProps = {
+		active: true,
+		position: 'top-left',
+		onClick: noop
+	};
 
-	getDefaultProps: function() {
-		return {
-			active: true,
-			position: 'top-left',
-			onClick: function() {}
-		};
-	},
+	constructor( props ) {
+		super( props );
+	}
 
-	getIconElement: function() {
-		if ( this.props.icon ) {
-			return <span className={ 'noticon noticon-' + this.props.icon } />;
+	getIconElement() {
```

Comment:
Thanks for the thoughts folks, but I'm removing that method anyway - as @aduth mentioned, it's simple enough to not require a specific method.